app/styles/messaging-styles.css

```diff
@@ -3,6 +3,10 @@
   rounded-2xl px-6 py-4 shadow-md;
 }
 
+.preserve-whitespace {
+  white-space: pre-wrap;
```

Comment:
Nice one! Really elegant solution.

To keep in line with the current formatting of our code and avoid editing an extra file we should apply the corresponding tailwind class

That would be `whitespace-pre-wrap`